src/kprofile.h

```diff
@@ -28,7 +29,15 @@ public slots:
 #else
     void startDownload(QWebEngineDownloadRequest*);
 #endif
-    void downloadFinished();
+
+private slots:
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+    void saveFile(QWebEngineDownloadItem*);
+    void openFile(QWebEngineDownloadItem*);
+#else
+    void saveFile(QWebEngineDownloadRequest*);
+    void openFile(QWebEngineDownloadRequest*);
+#endif
```

Comment:
You can minimize Qt 5-vs-6 conditional compilation (including the old code) as follows:

```c++
#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
typedef WebEngineDownloadType QWebEngineDownloadItem;
#else
typedef WebEngineDownloadType QWebEngineDownloadRequest;
#endif
```

Reply:
Very neat! Thank you
(Fun fact: I spent an embarrassing amount of time deciding if I should use "using" or "typedef" :>)

Reply:
My bad. What a silly mistake.